test/discr/discr_ops_test.py

```diff
@@ -0,0 +1,218 @@
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Unit tests for `discr_ops`."""
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+standard_library.install_aliases()
+
+import pytest
+import numpy as np
+
+import odl
+from odl.discr.discr_ops import _SUPPORTED_RESIZE_PAD_MODES
+from odl.util.testutils import almost_equal, example_element
+
+
+# --- ResizingOperator --- #
+
+
+def test_resizing_op_init():
+
+    # Test if the different init patterns run
+
+    space = odl.uniform_discr([0, -1], [1, 1], (10, 5))
+    res_space = odl.uniform_discr([0, -3], [2, 3], (20, 15))
+
+    odl.ResizingOperator(space, res_space)
+    odl.ResizingOperator(space, ran_shp=(20, 15))
+    odl.ResizingOperator(space, ran_shp=(20, 15), offset=(0, 5))
+    odl.ResizingOperator(space, ran_shp=(20, 15), pad_mode='symmetric')
+    odl.ResizingOperator(space, ran_shp=(20, 15), pad_const=1.0)
+    odl.ResizingOperator(space, ran_shp=(20, 15), pad_const=1.0,
+                         discr_kwargs={'nodes_on_bdry': True})
+
+
+def test_resizing_op_raise():
+
+    # domain not a uniformely discretized Lp
+    with pytest.raises(TypeError):
+        odl.ResizingOperator(odl.rn(5), ran_shp=(10,))
+
+    grid = odl.TensorGrid([0, 2, 3])
+    part = odl.RectPartition(odl.IntervalProd(0, 3), grid)
+    fspace = odl.FunctionSpace(odl.IntervalProd(0, 3))
+    dspace = odl.rn(3)
+    space = odl.DiscreteLp(fspace, part, dspace)
+    with pytest.raises(ValueError):
+        odl.ResizingOperator(space, ran_shp=(10,))
+
+    # different cell sides in domain and range
+    space = odl.uniform_discr(0, 1, 10)
+    res_space = odl.uniform_discr(0, 1, 15)
+    with pytest.raises(ValueError):
+        odl.ResizingOperator(space, res_space)
+
+    # non-integer multiple of cell sides used as shift (grid of the
+    # resized space shifted)
+    space = odl.uniform_discr(0, 1, 5)
+    res_space = odl.uniform_discr(-0.5, 1.5, 10)
+    with pytest.raises(ValueError):
+        odl.ResizingOperator(space, res_space)
+
+    # need either range or ran_shp
+    with pytest.raises(ValueError):
+        odl.ResizingOperator(space)
+
+    # offset cannot be combined with range
+    space = odl.uniform_discr([0, -1], [1, 1], (10, 5))
+    res_space = odl.uniform_discr([0, -3], [2, 3], (20, 15))
+    with pytest.raises(ValueError):
+        odl.ResizingOperator(space, res_space, offset=(0, 0))
+
+    # bad pad_mode
+    with pytest.raises(ValueError):
+        odl.ResizingOperator(space, res_space, pad_mode='something')
+
+
+dtype_params = ['float64', 'complex64']
+dtype_ids = [' dtype = {} '.format(p) for p in dtype_params]
+
+
+@pytest.fixture(scope="module", ids=dtype_ids, params=dtype_params)
+def dtype(request):
+    return request.param
+
+
+def test_resizing_op_properties(dtype):
```

Comment:
Should also use `fn_impl` to test various fn, and use `odl.FN_IMPLS[fn_impl].available_dtypes()` to find dtypes.